From fec8216f8d34c06083e22609823904bd95172fc9 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Fri, 13 Jan 2023 09:19:09 +0100
Subject: [PATCH 252/841] feat(stm32mp1): add PDEF_EXTS management

Manage optional ddr-fw option in cert tool for STM32MP1.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 4a5eca6cdf0fb1a5cf728a46276c218aeb120e0c feat(stm32mp1): add PDEF_EXTS management]
Change-Id: I8e32694cba116a83dc26211cd54e8c58abf3ebe1
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/285748
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Lionel DEBIEVE <lionel.debieve@foss.st.com>
---
 plat/st/stm32mp1/cert_create_tbbr.mk |  3 ++-
 plat/st/stm32mp1/stm32mp1_tbb_cert.c | 11 ++++++++++-
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/plat/st/stm32mp1/cert_create_tbbr.mk b/plat/st/stm32mp1/cert_create_tbbr.mk
index 5b1a3ed94..dd402ec2a 100644
--- a/plat/st/stm32mp1/cert_create_tbbr.mk
+++ b/plat/st/stm32mp1/cert_create_tbbr.mk
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2022, STMicroelectronics - All Rights Reserved
+# Copyright (c) 2022-2023, STMicroelectronics - All Rights Reserved
 #
 # SPDX-License-Identifier: BSD-3-Clause
 #
@@ -7,6 +7,7 @@
 # Override TBBR Cert to update generic certificate
 
 $(eval $(call add_define,PDEF_CERTS))
+$(eval $(call add_define,PDEF_EXTS))
 
 PLAT_INCLUDE	+= -I${PLAT_DIR}include
 
diff --git a/plat/st/stm32mp1/stm32mp1_tbb_cert.c b/plat/st/stm32mp1/stm32mp1_tbb_cert.c
index 0e77397c8..69d03bd9b 100644
--- a/plat/st/stm32mp1/stm32mp1_tbb_cert.c
+++ b/plat/st/stm32mp1/stm32mp1_tbb_cert.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2015-2022, ARM Limited and Contributors. All rights reserved.
+ * Copyright (c) 2015-2023, ARM Limited and Contributors. All rights reserved.
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -35,3 +35,12 @@ static cert_t stm32mp1_tbb_certs[] = {
 };
 
 PLAT_REGISTER_COT(stm32mp1_tbb_certs);
+
+static ext_t stm32mp1_tbb_ext[] = {
+	[0] = {
+		.opt = "ddr-fw",
+		.optional = 1
+	}
+};
+
+PLAT_REGISTER_EXTENSIONS(stm32mp1_tbb_ext);
-- 
2.47.2

